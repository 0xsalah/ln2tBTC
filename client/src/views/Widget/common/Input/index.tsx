import React, { useRef, useState } from 'react'

import { Hint, Tooltip } from '../Tooltip'

interface InputProps {
    label?: string
    actionText?: string
    svgIcon?: React.ReactNode
    placeholder?: string
    value?: string
    type?: string
    display?: 'block' | 'inline'
    className?: string
    hint?: React.ReactNode
    error?: string
    onChange?: (ev: React.ChangeEvent<HTMLInputElement>) => void
    name?: string
    step?: number
    min?: number
}

const Input: React.FC<InputProps> = (props) => {
    const { label, actionText, svgIcon, placeholder, value, type, className, hint, error, onChange, name } = props
    let questionMarkElement = useRef<HTMLSpanElement>(null);
    const [displayTooltip, setDisplayTooltip] = useState(false)

    return (
        <>
            <div className={`exchange__column ${className ?? ''}`}>
                <div className={`form-block ${error ? 'is_error' : ''}`}>
                    {label &&
                        <label className="form-block__label">
                            <span>
                                {label}
                            </span>
                            {hint &&
                                <Hint ref={questionMarkElement} onClick={() => {
                                    setDisplayTooltip(true)
                                }} />
                            }
                        </label>
                    }
                    <div className="form-block__box">
                        <input aria-label="input" step={props.step} min={props.min} name={name} onChange={onChange} type={type} className="form-block__input form-control" value={value} placeholder={placeholder} />
                        {actionText && <div className="form-block__unit">{actionText}</div>}
                        {svgIcon &&
                            <div className="form-block__icon">
                                {svgIcon}
                            </div>
                        }
                    </div>
                    {error &&
                        <div className="form-block__message">{error}</div>
                    }
                </div>
            </div>
            {
                <Tooltip active={displayTooltip} hintButton={questionMarkElement} onDismiss={() => setDisplayTooltip(false)} >
                    {hint}
                </Tooltip>
            }
        </>
    )
}


export default Input