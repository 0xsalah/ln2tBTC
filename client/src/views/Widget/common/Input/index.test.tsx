import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Input from ".";

test("snapshot of component rendered with no label", () => {
  const component = render(<Input />);
  expect(component.baseElement).toMatchSnapshot();
});

test("snapshot of component rendered with label", () => {
  const component = render(<Input label="Test" />);
  expect(component.baseElement).toMatchSnapshot();
});

test("snapshot of component rendered with hint", () => {
  const component = render(<Input label="Test" hint={<span>Hint</span>} />);
  expect(component.baseElement).toMatchSnapshot();
});

test("snapshot of component rendered with actionText", () => {
  const component = render(<Input label="Test" actionText="MAX" />);
  expect(component.baseElement).toMatchSnapshot();
});

test("snapshot of component rendered with svgIcon", () => {
  const component = render(
    <Input
      label="Test"
      svgIcon={
        <svg>
          <symbol id="icon-clock" viewBox="0 0 47 47">
            <circle
              cx="23.5"
              cy="23.5"
              r="22"
              strokeWidth="1.5"
              strokeLinecap="round"
              strokeLinejoin="round"
            />
            <path
              d="M23.5 9.5V23.5H37.5"
              strokeWidth="1.5"
              strokeLinecap="round"
              strokeLinejoin="round"
            />
          </symbol>
        </svg>
      }
    />
  );
  expect(component.baseElement).toMatchSnapshot();
});
