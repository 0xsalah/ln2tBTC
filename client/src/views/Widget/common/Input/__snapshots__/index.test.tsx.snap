// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot of component rendered with actionText 1`] = `
<body>
  <div>
    <div
      class="exchange__column "
    >
      <div
        class="form-block "
      >
        <label
          class="form-block__label"
        >
          <span>
            Test
          </span>
        </label>
        <div
          class="form-block__box"
        >
          <input
            aria-label="input"
            class="form-block__input form-control"
            value=""
          />
          <div
            class="form-block__unit"
          >
            MAX
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`snapshot of component rendered with hint 1`] = `
<body>
  <div>
    <div
      class="exchange__column "
    >
      <div
        class="form-block "
      >
        <label
          class="form-block__label"
        >
          <span>
            Test
          </span>
          <span
            class="tooltip-help"
          >
            <svg
              class="icon icon-question"
            >
              <use
                xlink:href="icons.svg#icon-question"
              />
            </svg>
          </span>
        </label>
        <div
          class="form-block__box"
        >
          <input
            aria-label="input"
            class="form-block__input form-control"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`snapshot of component rendered with label 1`] = `
<body>
  <div>
    <div
      class="exchange__column "
    >
      <div
        class="form-block "
      >
        <label
          class="form-block__label"
        >
          <span>
            Test
          </span>
        </label>
        <div
          class="form-block__box"
        >
          <input
            aria-label="input"
            class="form-block__input form-control"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`snapshot of component rendered with no label 1`] = `
<body>
  <div>
    <div
      class="exchange__column "
    >
      <div
        class="form-block "
      >
        <div
          class="form-block__box"
        >
          <input
            aria-label="input"
            class="form-block__input form-control"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`snapshot of component rendered with svgIcon 1`] = `
<body>
  <div>
    <div
      class="exchange__column "
    >
      <div
        class="form-block "
      >
        <label
          class="form-block__label"
        >
          <span>
            Test
          </span>
        </label>
        <div
          class="form-block__box"
        >
          <input
            aria-label="input"
            class="form-block__input form-control"
            value=""
          />
          <div
            class="form-block__icon"
          >
            <svg>
              <symbol
                id="icon-clock"
                viewBox="0 0 47 47"
              >
                <circle
                  cx="23.5"
                  cy="23.5"
                  r="22"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="1.5"
                />
                <path
                  d="M23.5 9.5V23.5H37.5"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="1.5"
                />
              </symbol>
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
